lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4085,10 +4171,14 @@ ParseResult FIRStmtParser::parseSeqMem() {
 
   if (parseId(id, "expected smem name") ||
       parseToken(FIRToken::colon, "expected ':' in smem") ||
-      parseType(type, "expected smem type") || parseOptionalRUW(ruw) ||
-      parseOptionalInfo())
+      parseType(type, "expected smem type"))
     return failure();
 
+  if (consumeIf(FIRToken::comma)) {
+    if (parseRUW(ruw) || parseOptionalInfo())
+      return failure();
+  }
```

Comment:
I think parseOptionalInfo needs to be called regardless of RUW.

Reply:
Yea, this both allows a dangling comma and doesn't properly parse optional parts.  I think most consumeIf calls I saw are probably wrong.

Reply:
Thanks for the catch, @uenoku.

@darthscsi I'm not sure what you mean. Feel free to add more information. If you have an example that should be added to the parse-errors file, that would be great.